datafusion/src/physical_plan/hash_join.rs

```diff
@@ -781,7 +782,7 @@ fn equal_rows(
             DataType::Float32 => equal_rows_elem!(Float32Array, l, r, left, right),
             DataType::Float64 => equal_rows_elem!(Float64Array, l, r, left, right),
             DataType::Timestamp(_, None) => {
-                equal_rows_elem!(Int64Array, l, r, left, right)
+                equal_rows_elem!(TimestampNanosecondArray, l, r, left, right)
```

Comment:
I think we also have to match on nanosecond unit in this case here (and on other units as well)

Reply:
oh yeah, good catch :+1:

Reply:
I think other units need to first support the data types of these units in the Schema. And there is an issue about join on the `Time64(TimeUnit::Millisecond)` columns. I am analyzing this issue, and then I will open an issue and fix it.

![image](https://user-images.githubusercontent.com/25944814/134839707-e66c795e-f2a6-458d-a2a0-e7b8ca8510f9.png)